capa/features/address.py

```diff
@@ -36,6 +36,28 @@ def __hash__(self):
         return int.__hash__(self)
 
 
+class DynamicAddress(Address):
+    """an address from a dynamic analysis trace"""
+
+    def __init__(self, id_: int, return_address: int):
+        assert id_ >= 0
+        assert return_address >= 0
+        self.id = id_
+        self.return_address = return_address
```

Comment:
the _id_ field should make it easier to extract api sequences in the future. I am not sure about the name of the return_address field however, isn't that address of the call instruction (the one before the return address?), not sure if that's a big change anyways.

Reply:
While it's called `caller`, the value is actually the location returned to. I see that `return_address` can be confusing though...
<img width="699" alt="2023-07-07_12-11-23_ida64" src="https://github.com/mandiant/capa/assets/17606537/0c828134-0d26-4c65-b538-d76ad781d891">

If you think we should stick to `caller`, I'm fine with that.

Reply:
ah I see, in that case I think we should stick to _return_address_ then.